### ISSUE TITLE
Merge into main package

### ISSUE THREAD

**Issue by nuernbergerA** [KristofMorva/laravel-ide-macros#13]
@KristofMorva maybe we can work together on a PR to merge your work into @barryvdh's package.

ther should already be support for macros.

but it doesn't work quit well.
barryvdh/laravel-ide-helper#531

**Comment by KristofMorva** [comment]
Hey @nuernbergerA, I didn't know the original package has macro support too.

Unfortunately it has been a long time since I was in the web development field, so I'm afraid I wouldn't be able to assist in the actual development effectively. However, of course, I'm open to discussions about providing a single package that integrates these together, if this package does actually solve whatever issue the other package has.

Before anyone starts working on such integration, it might be worth to also ask @barryvdh about it, to verify if it's doable properly, and if the logic in this package does not introduce the same, or other bugs.

Alternatively, if for some reason it would work better than the implementation in the original package, I can also grant @barryvdh full repository rights for this package, so he's able to maintain it, and it can simply be a dependency for https://github.com/barryvdh/laravel-ide-helper.

Cheers!

**Comment by willsmanley** [comment]
Are there any ongoing efforts to merge the packages?

**Comment by KristofMorva** [comment]
@willsmanley Sadly I don't know of such initiative.

**Comment by mortenscheel** [comment]
I've looked into how the original package currently handles macros, and here's what I learned:

The original package can only detect macros on classes that are registered as an alias in `config/app.php`.
So if you add a macro to `\Illuminate\Routing\Route` it will not be detected, but if you add it to `Illuminate\Support\Facades\Route` it does work (because the facade is registered as an alias).

All macro methods are registered as static methods, so the IDE doesn't know they're available on the object *instance* like when you use `@instantiated` with this package.

Parameter types and return types are not detected from the macro function's signature (which this package does). But if the macro definition has a docblock, that does get copied to the _ide_helper.php file.
